Make @material-ui/core a peer dependency

luzzif opened the pull request:
## Description
Material UI complains when multiple library instances are initialized in a single application.
The issue can be easily solved by making `@material-ui/core` a peer dependency plus development dependency, instead of having it as a production dependency only.

luzzif commented:
IMHO this is pretty important, have a look [here](https://material-ui.com/getting-started/faq#i-have-several-instances-of-styles-on-the-page).

It's a pretty trivial PR too, I would love for this to get approved fast.

nareshbhatia commented:
It appears that you have only moved material-ui/core to peerDependencies. You should also specify it in devDependencies. This will have the advantage that you can easily do local development in the repository, while not including the bundle in the build.

Also I would like to see zero `dependencies`. All dependencies should be moved to `devDependencies` and `peerDependencies`.

I have the same use case with my [React Force](https://github.com/nareshbhatia/react-force) library that depends on Material UI. See this [package.json](https://github.com/nareshbhatia/react-force/blob/master/package.json) as an example.

luzzif commented:
Well I have moved `@material-ui/core` in `devDependencies` right from the start, so I don't know what you are talking about.

As far as making all dependencies peers, I moved `@material-ui/core` just because I know that multiple instances in the same application don't play too well. 
IMHO there are some dependencies that make sense as prod dependencies and not peers.

nareshbhatia commented:
@luzzif, good that `@material-ui/core` is already in `devDependencies` - it's my miss on my part.

As for other dependencies, I feel that only those dependencies should be included that no one else is likely to include in their own apps. For example, I already have `@material-ui/pickers` & `classnames` in my app.

luzzif commented:
Yeah, it's definitely debatable, but personally in my app I don't directly use `classnames`, so I wouldn't like my `package.json` to get cluttered by it.

I would love to see in my `package.json` only those dependencies which I directly use or which I directly need (the only exceptions being dependencies which create problems when multiple instances are found in the same app: those should always be peers).

P.S.: sorry if I sounded jerky when I told you that I already had added `@material-ui/core` to the `devDependencies`, I obviously didn't mean to.

nareshbhatia commented:
As a side note, it might be good to use storybook to develop & demo the various features. It eliminates the need for creating CodeSandboxes and/or separate apps for each feature. Just a thought.

nareshbhatia commented:
nw, @luzzif - no offense taken :-)

I get you point about adding more libraries into peer dependencies. It's a double-edged sword!

Personally, I would like to use material-table only for its sorting feature. I would like to see it come to a point where including it in my app does not add too much weight. Until then I will simply add a poor-man's sort on top of Material UI's table (which adds only 1K gzipped to my app).

luzzif commented:
Can someone approve this?

victor-accarini commented:
@mbrn This is pretty important see #773, #801 and #804.

luzzif commented:
It is important indeed. It baffles me how much time a little PR like this takes to get approved...

victor-accarini commented:
I believe the author was away from the project last month, it will probably take a while until he get up to date with the PRs. Last time he merged mine in less than a day, he is quite fast 👍

mbrn commented:
> I believe the author was away from the project last month, it will probably take a while until he get up to date with the PRs. Last time he merged mine in less than a day, he is quite fast 👍

Thank @victor-accarini . For now i will merge PRs only. Then we will work on issues with @emrekara37.

luzzif commented:
So is this going to be approved? Material Table totally breaks the page's styles when used with the last few versions of Material UI. It's pretty bad...

eps1lon commented:
For anybody having issues feel free to post your lockfiles here and I should be able to provide you with a patch that fixes those issues.
